swan-lake/development-tutorials/deployment-guide/virtual-machine-deployment.md

```diff
@@ -15,64 +25,155 @@ intro: If you prefer to execute Ballerina services directly on virtual machines
 
 3. Install any other dependencies your Ballerina service might need (e.g., database drivers, libraries).
 
-4. Write Ballerina code.
-
-      ```ballerina
-         import ballerina/http;
-            
-         service / on new http:Listener(9090) {
-             resource function get sayHello() returns string {
-                 return "Hello World!";
-             }
-         }
-      ```
+4. Implement the business logic in the Ballerina.
+
    > **Tip:** Organize the Project: Follow the standard Ballerina project structure, with a `Ballerina.toml` file and source organized in modules if needed. You can visit the [Organize Ballerina code](https://ballerina.io/learn/organize-ballerina-code/) for detailed information about packages and how you can manage the growth of your source code.
-   
-5. Build the Ballerina service.
-    
-    - Compile the service. This command creates an executable `.jar` file in the `target/bin` directory.
-      ```yaml
-      $ bal build
-      ```
+
+5. Build the Ballerina service. This command compiles the Ballerina application to produce the target artifacts. The default build command creates an executable `.jar` file in the `target/bin` directory.
+
+   ```bash
+   $ bal build
+   ```
+
    > **Tip**: You can use GraalVM builds to compile Ballerina apps into native binaries, ensuring faster startup and lower memory usage compared to the JVM.
    >
-   > ```
+   > ```bash
    > $ bal build --graalvm
    > ```
+   >
    > See [Build the GraalVM executable](https://ballerina.io/learn/build-the-executable-locally/) for more information.
 
-## CI/CD integration with VMs
+6. Publish the artifacts to the registry.
```

Comment:
Though the title says CI, the automation of building, testing, and publishing is unclear. Shall we improve this part to explain how CI happens in a decentralized deployment?

Reply:
Fixed via [4c1f3f0](https://github.com/ballerina-platform/ballerina-dev-website/pull/9800/commits/4c1f3f0911c65709821f49e7f6ae08d244becf46)